models/articles.models.js

```diff
@@ -1,20 +1,51 @@
 const db = require("../db/connection");
 
-exports.fetchArticles = () => {
-  return db
-    .query(
-      `SELECT articles.*, COUNT(comments.comment_id) AS comment_count
+exports.fetchArticles = (
+  sort_by = "created_at",
+  order = "DESC",
+  topic = undefined
+) => {
+  if (topic !== undefined && order !== undefined && sort_by !== undefined) {
+    return db
+      .query(
+        `SELECT articles.*, COUNT(comments.comment_id) AS comment_count
+   FROM articles LEFT JOIN comments
+   ON articles.article_id = comments.article_id
+   WHERE topic = $1
+   GROUP BY articles.article_id ORDER BY ${sort_by} ${order};`,
+        [topic]
+      )
+      .then(({ rows }) => {
+        if (rows.length === 0) {
+          return Promise.reject({ status: 404, msg: "404 Not Found" });
+        } else {
+          const deleteUnuseItem = rows.map((each) => {
+            delete each.body;
+            return each;
+          });
+          return deleteUnuseItem;
+        }
+      });
+  } else if (topic === undefined) {
+    return db
+      .query(
+        `SELECT articles.*, COUNT(comments.comment_id) AS comment_count
       FROM articles LEFT JOIN comments
       ON articles.article_id = comments.article_id
-      GROUP BY articles.article_id ORDER BY articles.created_at DESC;`
-    )
-    .then(({ rows }) => {
-      const deleteUnuseItem = rows.map((each) => {
-        delete each.body;
-        return each;
+      GROUP BY articles.article_id ORDER BY ${sort_by} ${order};`
```

Comment:
Here (and on line 15) we leave ourselves open to SQL injection. We can't use the $1 syntax to protect from this in this instance, so we need another way to make sure the user input here is safe.
We need to include a list of valid (greenlisted) sort_by options, and order options, and check the user input is included in those lists before proceeding with the query. We need to provide the same list for the topics as well (though this list can be populated by the fetchTopics function you've already written).